pkg/cloudprovider/provider/digitalocean/provider.go

```diff
@@ -188,6 +189,17 @@ func (p *provider) Validate(ctx context.Context, spec clusterv1alpha1.MachineSpe
 		return fmt.Errorf("invalid operating system specified %q: %w", pc.OperatingSystem, err)
 	}
 
+	switch f := pc.Network.GetIPFamily(); f {
+	case util.Unspecified, util.IPv4:
+	// noop
+	case util.IPv6:
+		return fmt.Errorf(util.ErrIPv6OnlyUnsupported)
+	case util.DualStack:
+		// noop
+	default:
+		return fmt.Errorf(util.ErrUnknownNetworkFamily, f)
+	}
```

Comment:
Do you really need `switch` here? All I could garner is that you need:

```
if pc.Network.GetIPFamily() == util.IPv6 {
     return fmt.Errorf(util.ErrIPv6OnlyUnsupported)
}
```

We'll never reach the `default` condition imo.

Reply:
If you mistype `IPv6` as `Ipv6` you can reach the default condition.

Reply:
```
// IPFamily IPv4 | IPv6 | IPv4+IPv6.
type IPFamily string

const (
	Unspecified IPFamily = "" // interpreted as IPv4
	IPv4        IPFamily = "IPv4"
	IPv6        IPFamily = "IPv6"
	DualStack   IPFamily = "IPv4+IPv6"
)
```

I assumed that marshaling would have errored out since it's not an acceptable value for the type. But I guess it doesn't.

Can we add a validation directly in the webhook for this? Just to check if IPFamily has a valid value, irrespective of the cloud provider. That'd make our life so much easier.